data/src/time.rs

```diff
@@ -14,4 +14,13 @@ impl Posix {
 
         Self(nanos_since_epoch)
     }
+
+    pub fn parse_rfc3339(date: impl AsRef<str>) -> Option<Self> {
+        let datetime = chrono::DateTime::parse_from_rfc3339(date.as_ref()).ok()?;
```

Comment:
I say we update Message to hold a DateTime<Utc> instead of posix. Since we will most likely need to format this into human readable anyways, it makes sense.

We can always make the serde implementation go into / from u64 timestamp for compactness.